Fabric.Authorization.Client/AuthorizationClient.cs

```diff
@@ -34,16 +35,41 @@ public AuthorizationClient(HttpClient client)
 
         public async Task<UserPermissionsApiModel> GetPermissionsForCurrentUser(string accessToken)
         {
-            CheckIfNull(accessToken, nameof(accessToken));
+            return await GetPermissionsForCurrentUser(accessToken, null, null).ConfigureAwait(false);
+        }
+
+        public async Task<UserPermissionsApiModel> GetPermissionsForCurrentUser(string accessToken, string grain, string securableItem)
+        {
+            CheckIfStringNullOrEmpty(accessToken, nameof(accessToken));
 
-            var route = new UserRouteBuilder().UserPermissionsRoute;
+            var route = new UserRouteBuilder()
+                            .Grain(grain)
+                            .SecurableItem(securableItem)
+                            .UserPermissionsRoute;
             var message = new HttpRequestMessage(HttpMethod.Get, route)
                 .AddBearerToken(accessToken)
                 .AddAcceptHeader(ClientConstants.ApplicationJson);
 
             return await SendAndParseJson<UserPermissionsApiModel>(message).ConfigureAwait(false);
         }
-        
+
+        public async Task<bool> DoesUserHavePermission(string accessToken, string permission)
+        {

```

Comment:
General comment and I'm not sure what others' thoughts are, but should we just pass through whatever the user passes in and let the API do the validation of parameters? So essentially remove the `CheckIf*` calls?

Reply:
I think checking for null/empty strings is fine - it will help to keep clients from having to go to the server to find out that they didn't pass an access token, which reduces the time it takes to get an error response and lowers the load on the service itself. I wouldn't be any deeper validation on the client though, as we want to keep that logic with the API itself.

Reply:
That works.

Reply:
Agreed